src/meteors/shap/explainer.py

```diff
@@ -0,0 +1,196 @@
+
+        if len(explanation_shape) == len(data_shape) + 1:
+            # the explanation has one more dimension than the data - the multitarget explanation, no local explanation
+            if explanation_shape[:-1] == data_shape:
+                logger.debug("Detected explanation for multiple targets based on the explanation shape validation.")
+                return
+            else:
+                raise ShapeMismatchError(
+                    f"Shape of the explanations does not match the shape of the input data. Expected {data_shape}, but got {self.explanations.shape}"
+                )
+        elif len(explanation_shape) == len(data_shape):
+            if explanation_shape[-2] == data_shape[-1] and data_shape[0] == 1:
+                logger.debug("Detected explanation for multiple targets based on the explanation shape validation.")
+                logger.debug("This is a local explanation for multiple targets. Adding a new axis to the explanation.")
+                self.explanations = add_dimension_to_local_explanation(self.explanations)
+                if self.explanations.shape[:-1] == data_shape:
+                    return
+        elif len(explanation_shape) == len(data_shape) - 1:
+            logger.debug("Detected local explanation for a single target. Removing the last axis from the explanation.")
+            self.explanations = add_dimension_to_local_explanation(self.explanations)
+            if self.explanations.shape == data_shape:
+                return
+
+        raise ShapeMismatchError(
+            f"Shape of the explanations does not match the shape of the input data. Expected {data_shape}, but got {explanation_shape}"
+        )
+
+    @model_validator(mode="after")
+    def validate_explanations(self) -> Self:
+        """
+        Validates the the explanations.
+
+        Returns:
+            Self: The instance of the class for method chaining.
+        """
+        self._validate_shapes()
+        return self
+
+    @property
+    def num_target_outputs(self) -> int:
+        """
+        Number of target dimensions in the explanations. It is equal to the number of outputs of the model. For instance,
+        in scope of the HYPERVIEW Challenge, the EAGLEEYES model has 4 outputs - it predicts 4 different soil parameters.
+        For explanations of this model, the `num_target_outputs` property will equal to 4.
+        Returns:
+            int: number of target dimensions in the explanations.
+        """
+        if self.explanations.shape == self.data.shape:
+            return 1
+        return self.explanations.values.shape[-1]
+
+    @property
+    def is_local_explanation(self) -> bool:
+        """check if the explanation is for a single observation - a local explanation.
+        Returns:
+            bool: True if the explanation is for a single observation, False otherwise.
+        """
+        if len(self.data.shape) == 1:
+            return True
```

Comment:
Add test for it as codecov marked it as not covered

Reply:
removed the line as it is impossible to execute it after data field validation